libraries/extractor/src/main/java/androidx/media3/extractor/mp4/BoxParser.java

```diff
@@ -114,6 +117,16 @@ public final class BoxParser {
   /** The magic signature for an Opus Identification header, as defined in RFC-7845. */
   private static final byte[] opusMagic = Util.getUtf8Bytes("OpusHead");
 
+  /** This is a hack to get the width and height from the video track into any vobsub
+   * subtitle tracks. It would be preferable to modify the vobsub track initialization
+   * data in parseTraks after all tracks have been parsed, but all of the structures
+   * involved are final so cannot be modified. These are default values that will be
+   * updated as video tracks are parsed, then used when a vobsub track is parsed.
+   * If the vobsub occurs before the video track, the default values here will be used.
+   */
+  private static int width = 720;
+  private static int height = 576;
```

Comment:
I'm afraid we can't accept this PR with this static mutable state. It's quite possible that there could be two ExoPlayer instances in the same JVM, both handling VobSub subtitles in MP4 files, and this static state would result in very confusing "cross talk" between them.

I think you will need to re-visit this plumbing to remove the static mutability.

Reply:
I agree this is bad. Thanks for the feedback. I will try to work it in a different way.